applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/bluetooth/conn.h>
+
+#include <caf/events/ble_common_event.h>
+#include <caf/events/ble_smp_event.h>
+#include "config_event.h"
+#include "usb_event.h"
+#include <caf/events/button_event.h>
+#include <caf/events/power_event.h>
+
+#define MODULE dvfs
+#include <caf/events/module_state_event.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_DVFS_LOG_LEVEL);
+
+#include "ld_dvfs_handler.h"
+
+#define DVFS_INIT_TIMEOUT		K_SECONDS(1)
+#define REG_CONN_INTERVAL_LLPM_MASK	0x0d00
+
+enum dvfs_states_list {
+	DVFS_STATE_DEFAULT,
+	DVFS_STATE_LLPM,
+	DVFS_STATE_USB,
+	DVFS_STATE_COUNT
+};
+
+/* Table containing the frequency settings for each state.
+ *It is tailored for nrf54h20dk and might need to be adjusted for other platforms.
+ */
+enum dvfs_frequency_setting dvfs_freq_table[DVFS_STATE_COUNT] = {
+	DVFS_FREQ_LOW,		/* DVFS_STATE_DEFAULT */
+	DVFS_FREQ_MEDLOW,	/* DVFS_STATE_LLPM */
+	DVFS_FREQ_HIGH		/* DVFS_STATE_USB */
```

Comment:
Consider switching to bitmasks for `MEDLOW` and `HIGH` states requests to speed up setting proper DVFS state on app_state_changes. IIRC, we agreed on this approach during our design meeting and I even pointed you to CAF Power Manager for an example.

Reply:
I don't recall it being said there but never mind.
Is this worth doing, because we don't control DVFS_FREQ_ macros and we would have to re declare them and we loose some memory and gain some performance.